checks/prometheus_check.py

```diff
@@ -284,12 +329,49 @@ def process_metric(self, message, send_histograms_buckets=True, custom_tags=None
         `send_histograms_buckets` is used to specify if yes or no you want to send the buckets as tagged values when dealing with histograms.
         """
         try:
+            # If targeted metric, store labels
+            if message.name in self.label_joins:
+                matching_label = self.label_joins[message.name]['label_to_match']
+                for metric in message.metric:
+                    labels_list = []
+                    matching_value = None
+                    for label in metric.label:
+                        if label.name == matching_label:
+                            matching_value = label.value
+                        elif label.name in self.label_joins[message.name]['labels_to_get']:
+                            labels_list.append((label.name, label.value))
+                    try:
+                        self._label_mapping[matching_label][matching_value] = labels_list
+                    except KeyError:
+                        if matching_value is not None:
+                            self._label_mapping[matching_label] = {matching_value: labels_list}
+
             if message.name in self.ignore_metrics:
```

Comment:
Do you think it's useful to move this condition at the top ?

Reply:
I placed it intentionally bellow because we might want to target a metric for joining label on it but ignoring it for sending (it's the case of the `kube_pod_info` for example)